### PR TITLE
OSD-11365 - Add test for techpreviewnoupgrade webhook

### PR DESCRIPTION
**Pull request by Tafhim**
Resolves [OSD-11365](https://issues.redhat.com//browse/OSD-11365)
Adds a test to verify the functionality implemented in [OSD-10922](https://issues.redhat.com//browse/OSD-10922)

**Context**: [OSD-10922](https://issues.redhat.com//browse/OSD-10922) implements a webhook to make sure the FeatureSet called *TechPreviewNoUpgrade* cannot be used in FeatureGates. It should be blocked using Managed Cluster Validating Webhooks. This test verifies that the functionality works properly.

I have put this PR on do-not-merge because there is another [PR](https://github.com/openshift/managed-cluster-validating-webhooks/pull/189) that needs to be merged first for this test to be valid.

### CODE REVIEW

File: pkg/e2e/verify/techPreviewNoUpgrade_webhook.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package verify
```
Comment (mrbarge) [inline pkg/e2e/verify/techPreviewNoUpgrade_webhook.go:1]: golang convention is for filenames to be entirely lowercase.

Comment (Tafhim) [reply pkg/e2e/verify/techPreviewNoUpgrade_webhook.go]: Renamed file to match the convention. @mrbarge

File: pkg/e2e/verify/techPreviewNoUpgrade_webhook.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			// first fetch the ResourceVersion
+			existingFeatureGate, err := h.Cfg().ConfigV1().FeatureGates().Get(context.TODO(), "cluster", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			existingFeatureGate.Spec.FeatureSet = "TechPreviewNoUpgrade"
+
+			_, err = h.Cfg().ConfigV1().FeatureGates().Update(context.TODO(), existingFeatureGate, metav1.UpdateOptions{})
+
+			Expect(err).To((HaveOccurred()))
+		})
```
[the remark this answers is not in the document]
Comment (Tafhim) [reply pkg/e2e/verify/techPreviewNoUpgrade_webhook.go]: Applied timeout explicitly @mrbarge

File: pkg/e2e/verify/techPreviewNoUpgrade_webhook.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+func init() {
+	alert.RegisterGinkgoAlert(techPreviewNoUpgradeWebhookTestName, "SD-SREP", "Tafhim Ul Islam", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(techPreviewNoUpgradeWebhookTestName, func() {
+
+	h := helper.New()
+
+	ginkgo.Context("techpreviewnoupgrade webhook", func() {
+		util.GinkgoIt("Webhook will not block CREATE requests if TechPreviewNoUpgrade FeatureGate does not exist", func() {
```
[the remark this answers is not in the document]
Comment (Tafhim) [reply pkg/e2e/verify/techPreviewNoUpgrade_webhook.go]: Updated test description @mrbarge

**Comment by Tafhim** [comment]
Removing hold because the [Blocking PR](https://github.com/openshift/managed-cluster-validating-webhooks/pull/189) has been merged.
/unhold

**Pull request merged**